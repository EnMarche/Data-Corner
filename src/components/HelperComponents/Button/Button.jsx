/* eslint-disable react/forbid-prop-types */
import React from 'react';
import { Button as MuiButton, makeStyles, createStyles } from '@material-ui/core';
import PropTypes from 'prop-types';

const useStyles = makeStyles((theme) => createStyles({
    root: {
        borderRadius: '8.35px',
        padding: theme.spacing(1, 2),
    },
}));

const Button = ({ children, buttonClasses, handleClick = () => {} }) => {
    const classes = useStyles();

    return (
        <MuiButton
            variant="contained"
            className={`${classes.root} ${buttonClasses}`}
            onClick={handleClick}
        >
            {children}
        </MuiButton>
    );
};

export default Button;

Button.propTypes = {
    children: PropTypes.node.isRequired,
    buttonClasses: PropTypes.string.isRequired,
    handleClick: PropTypes.func.isRequired,
};
