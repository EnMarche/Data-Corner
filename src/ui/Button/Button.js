import { Button as MuiButton } from '@mui/material'
import { makeStyles } from '@mui/styles'
import PropTypes from 'prop-types'

const useStyles = makeStyles(theme => ({
  root: {
    borderRadius: '8px',
    padding: theme.spacing(1, 2),
  },
}))

const Button = ({ children, buttonClasses, handleClick, disabled }) => {
  const classes = useStyles()

  return (
    <MuiButton
      variant="contained"
      className={`${classes.root} ${buttonClasses}`}
      onClick={handleClick}
      disabled={disabled}
    >
      {children}
    </MuiButton>
  )
}

export default Button

Button.defaultProp = {
  disabled: false,
}

Button.propTypes = {
  children: PropTypes.node.isRequired,
  buttonClasses: PropTypes.string.isRequired,
  handleClick: PropTypes.func.isRequired,
  disabled: PropTypes.bool,
}
