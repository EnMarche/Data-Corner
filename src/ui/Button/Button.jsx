import { Button as MuiButton, makeStyles, createStyles } from '@material-ui/core';
import PropTypes from 'prop-types';

const useStyles = makeStyles((theme) => createStyles({
    root: {
        borderRadius: '8.35px',
        padding: theme.spacing(1, 2),
    },
}));

const Button = ({
    children, buttonClasses, handleClick, disabled,
}) => {
    const classes = useStyles();

    return (
        <MuiButton
            variant="contained"
            className={`${classes.root} ${buttonClasses}`}
            onClick={handleClick}
            disabled={disabled}
        >
            {children}
        </MuiButton>
    );
};

export default Button;

Button.defaultProp = {
    disabled: true,
}

Button.propTypes = {
    children: PropTypes.node.isRequired,
    buttonClasses: PropTypes.string.isRequired,
    handleClick: PropTypes.func.isRequired,
    disabled: PropTypes.bool,
};
