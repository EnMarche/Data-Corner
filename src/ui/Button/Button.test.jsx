import { render } from '@testing-library/react'
import Button from './Button'

jest.mock('@material-ui/core', () => ({
    Button: ({ children }) => <div className="mui-button-mock">{ children }</div>,
    makeStyles: () => () => ({}),
}))

describe('Button', () => {
    it('displays button', () => {
        const mock = jest.fn()
        const { container } = render(
            <Button buttonClasses="fooClass" handleClick={mock}>
                Foo
            </Button>,
        )

        expect(container).toMatchSnapshot()
    })
})
