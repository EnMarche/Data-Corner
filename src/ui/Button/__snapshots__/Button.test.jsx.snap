// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button displays button 1`] = `
<div>
  <div
    class="mui-button-mock"
  >
    Foo
  </div>
</div>
`;
